inject/src/main/java/io/micronaut/context/DefaultBeanContext.java

```diff
@@ -934,6 +940,58 @@ protected <T> Stream<T> streamOfType(BeanResolutionContext resolutionContext, Cl
         return streamOfType(resolutionContext, Argument.of(beanType), qualifier);
     }
 
+    /**
+     * Obtains a map of beans of the given type and qualifier.
+     * @param resolutionContext  The resolution context
+     * @param beanType           The bean type
+     * @param qualifier          The qualifier
+     * @param <K>                The key type
+     * @param <V>                The bean type
+     * @return A map of beans, never {@code null}.
+     * @since 4.0.0
+     */
+    protected <V> @NonNull Map<String, V> mapOfType(@Nullable BeanResolutionContext resolutionContext, @NonNull Argument<V> beanType, @Nullable Qualifier<V> qualifier) {
```

Comment:
So If I call this with `mapOfType(MyInterface.class, Qualifiers.ANY)` and:
```
@Singleton
class A implements MyInterface {
}
@Singleton
class B implements MyInterface {
}
@Singleton
class C implements MyInterface {
}
```
I will get only one item in the map with `primary` key?

Reply:
Ok I see it should return `occurs for duplicate keys` because the stream collet wouldn't allow a conflict. It's still strange that this would be OK when only one implementation is defined but will fail with multiple. IMHO correct would be to fail even with one implementation that doesn't specify some criteria like not being named. 

`Multiple beans were found missing a qualifier resulting in duplicate keys` What kind of qualifier is it missing? 

It's confusing to me that in the Micronaut codebase there are two understandings of what qualifier means:
1. Bean context uses it as a filter
2. Bean definition uses is as a set of `@Qualifier`s defined